src/00_Core/Player/LinkStateDamage.cpp

```diff
@@ -1,16 +1,218 @@
+    unk1 = this->PlayerControlData_vfunc_14(data_ov000_020e5b00.mUnk_00);
+
+    // Back minor damage regeneration speed "後小ダメ再生速度"
+    const char *description4 = "\x8c\xE3\x8F\xAC\x83_\x83\x81\x8D\xC4\x90\xB6\x91\xAC\x93x\x00\x00\x00";
+
+    // Back minor damage start frame "後小ダメ開始フレーム"
+    const char *description5 = "\x8C\xE3\x8F\xAC\x83_\x83\x81\x8AJ\x8En\x83t\x83\x8c\x81[\x83\x80\x00\x00\x00";
+
+    // Back minor damage end frame "後小ダメ終了フレーム"
+    const char *description6 = "\x8C\xE3\x8F\xAC\x83_\x83\x81\x8FI\x97\xB9\x83t\x83\x8c\x81[\x83\x80\x00\x00\x00";
+
+    debugHierarchy = this->GetDebugHierarchy0();
+    debugHierarchy->vfunc_3c(id, &data_ov000_020e5b00.mUnk_04.x);
+    debugHierarchy->GetChildNode(1, description4, id, &data_ov000_020e5b00.mUnk_04.x, 8, 0, 0x2000, 0);
+    debugHierarchy = this->GetDebugHierarchy0();
+    debugHierarchy->vfunc_3c(id, &data_ov000_020e5b00.mUnk_04.y);
+    debugHierarchy->GetChildNode(1, description5, id, &data_ov000_020e5b00.mUnk_04.y, 8, 0, unk1 / 2, 0);
+    debugHierarchy = this->GetDebugHierarchy0();
+    debugHierarchy->vfunc_3c(id, &data_ov000_020e5b00.mUnk_04.z);
+    debugHierarchy->GetChildNode(1, description6, id, &data_ov000_020e5b00.mUnk_04.z, 8, unk1 / 2, unk1, 0);
+
+    // Number of tears [verb] "ビリビリ回数"
+    const char *description7 = "\x83r\x83\x8A\x83r\x83\x8A\x89\xF1\x90\x94\x00\x00\x00";
+
+    debugHierarchy = this->GetDebugHierarchy0();
+    debugHierarchy->GetChildNode(0, description7, id, &data_ov000_020e5aec, 6, 0, 0x14, 0);
+
+    // Tear [verb] playback speed "ビリビリ再生速度"
+    const char *description8 = "\x83r\x83\x8A\x83r\x83\x8A\x8D\xC4\x90\xB6\x91\xAC\x93x\x00\x00\x00";
+
+    debugHierarchy = this->GetDebugHierarchy0();
+    debugHierarchy->vfunc_3c(id, &data_ov000_020e5adc.mUnk_04.x);
+    debugHierarchy->GetChildNode(1, description8, id, &data_ov000_020e5adc.mUnk_04.x, 8, 0, 0x2000, 0);
+    this->GetDebugHierarchy0();
+    this->GetDebugHierarchy0();
+    this->GetDebugHierarchy0();
+    this->GetDebugHierarchy0();
+    this->GetDebugHierarchy0();
+    this->GetDebugHierarchy1();
+}
+
+void LinkStateDamage::OnStateEnter() {
+    func_ov00_020a8a4c(&data_ov000_020e5b10, 1);
+    func_ov00_020a8a4c(&data_ov000_020e5b20, 1);
+    func_ov00_020a8a4c(&data_ov000_020e5b30, 1);
+    func_ov00_020a8a4c(&data_ov000_020e5b40, 1);
+    func_ov00_020a8a4c(&data_ov000_020e5b50, 1);
+}
 
-void LinkStateDamage::vfunc_00() {}
-void LinkStateDamage::CreateDebugHierarchy() {}
-void LinkStateDamage::OnStateEnter() {}
 void LinkStateDamage::OnStateLeave(s32 param1) {}
-void LinkStateDamage::func_ov00_020ac9e4(unk32 param1) {}
-void LinkStateDamage::func_ov00_020aca50() {}
-void LinkStateDamage::func_ov00_020aca94() {}
-void LinkStateDamage::vfunc_30(unk32 param1) {}
+
+ARM void LinkStateDamage::func_ov00_020ac9e4(unk32 param1) {
+    bool unkBool = func_ov005_02110f50(this->mUnk_30, param1, this->mUnk_22, (u32 *) this->mUnk_b0);
+    if (!unkBool) {
```

Comment:
that's fake for sure, it should be
```suggestion
    if (!func_ov005_02110f50(this->mUnk_30, param1, this->mUnk_22, (u32 *) this->mUnk_b0)) {
```